tests/e2e/fixtures/kserve_denpendencies.py

```diff
@@ -0,0 +1,186 @@
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+@pytest.fixture(scope="class")
+def kserve_secret(metadata, cluster, region, request):
+    metadata_key = "aws-secret"
+
+    def on_create():
+        secret_config = load_yaml_file(SECRET_CONFIG_FILE)
+        secret_config["metadata"]["annotations"]["serving.kserve.io/s3-region"] = region
+        write_yaml_file(secret_config, SECRET_CONFIG_FILE)
+        kubectl_apply(SECRET_CONFIG_FILE)
+        #patch secret to IRSA in prifle namespace
+        exec_shell("kubectl patch serviceaccount aws-sa -n kubeflow-example-com -p '{\"secrets\": [{\"name\": \"aws-secret\"}]}'")
+        
+
+    def on_delete():
+        cmd = f"kubectl delete secret aws-secret -n kubeflow-example-com".split()
+        subprocess.call(cmd)
+
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+@pytest.fixture(scope="class")
+def kserve_inference_service(metadata, cluster, region, request):
+    metadata_key = "kserve-inference-service"
+
+    def on_create():
+        # Namespaces created by the Kubeflow profile controller have a missing authorization policy that
+        # prevents the KServe predictor and transformer from working.
+        # https://github.com/kserve/kserve/issues/1558
+        # https://github.com/kubeflow/kubeflow/issues/5965
+        # https://github.com/awslabs/kubeflow-manifests/issues/82#issuecomment-1068641378
+
+        print("creating allow-predictor AuthorizationPolicy...")
+        kubectl_apply(AUTHORIZATION_POLICY_CONFIG_FILE)
+
+        print("creating inference service...")
+        bucket_name = metadata.get("s3-bucket")
+        
+        inference_config = load_yaml_file(INFERENCE_CONFIG_FILE)
+        inference_config["spec"]["predictor"]["model"]["storageUri"] = f"s3://{bucket_name}/saved_model_half_plus_two/"
+        write_yaml_file(inference_config, INFERENCE_CONFIG_FILE)
+        kubectl_apply(INFERENCE_CONFIG_FILE)
+        wait_for_inference(cluster, region)
+
+    def on_delete():
+        cmd = f"kubectl delete -f {INFERENCE_CONFIG_FILE}".split()
+        subprocess.call(cmd)
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+
+    
+
+
+def wait_for_inference(cluster_name: str, region: str):
+        logger.info("waiting for inference service creation ...")
+
+        def callback():
+            inference_service = get_inference_service(cluster_name, region, namespace="kubeflow-example-com")
+            
+            assert inference_service.get("status") is not None
+            conditions = inference_service["status"]["conditions"]
+            for condition in conditions:
```

Comment:
Is there documentation for all the different type of condition["type"]?  Seems like we are checking different types (e.g,"IngressReady,PredictorReady") but we have the same assert statement for all them? Are there other types of conditions that we will not be checking?

Reply:
It's equivalent to check the inferenceservice status to be READY https://awslabs.github.io/kubeflow-manifests/main/docs/component-guides/kserve/tutorial/#check-inferenceservice-status

Reply:
I guess what i am saying is that your are looping through a conditions list, and then running if statements on those conditions but it seems no matter what condition you have you have the same response? So does it even matter to specifically check those fields and since you are just running through this for-loop and it doesn't matter if you have all of the four cases you check it may still succeeed even if you only have one of them.

Reply:
There are four types in the conditions and if one of them is not "True" the assert statement will fail.